src/rockstor/storageadmin/static/storageadmin/js/router.js

```diff
@@ -757,7 +758,30 @@ var AppRouter = Backbone.Router.extend({
         this.currentLayout = new LogsView();
         $('#maincontent').empty();
         $('#maincontent').append(this.currentLayout.render().el);
-    }
+    },
+	
+	showShell: function() {
+		//Special router function for shell
+		//We fetch shellinaboxd service model and check for detach option
+		//If not present or false we go with normal layout to maincontent
+		//otherwhise (popup enabled) we open it in a new detached window
+		var _this = this;
+		_this.shell_service = new Service({name: 'shellinaboxd'});
+		_this.shell_service.fetch({
+		success: function(collection){
+			var config = JSON.parse(collection.get('config'));
+			if ('detach' in config && config.detach){
```

Comment:
This is awesome @MFlyer! Nicely done. I did a quick test and noticed a minor issue. If the service is not configured and you click on System -> System-Shell, nothing happens in the UI and there's an error in the browser-console because the service config is null. It would be good UX to prompt the user if shellinabox is not ready when they click on there. For example, if it's not configured, then promt them to configure and if the service is not started, give them the option to do so.